cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternals.qll

```diff
@@ -74,9 +64,10 @@ private class DefOrUse extends TDefOrUse {
   /** Gets the block of this definition or use. */
   abstract IRBlock getBlock();
 
-  /** Holds if this definition or use has rank `rank` in block `block`. */
-  cached
-  final predicate hasRankInBlock(IRBlock block, int rnk) { rnk = getRank(this, block) }
+  /** Holds if this definition or use has index `index` in block `block`. */
+  final predicate hasIndexInBlock(IRBlock block, int index) {
```

Comment:
OOI: Why has the cached annotation been removed?

Reply:
The annotation was there to force the work done by `getRank` to be cached, but the new predicate uses the `IRBlock::getInstruction` member predicate, which itself uses a cached [nonmember `getInstruction` predicate](https://github.com/github/codeql/blob/b3cb250ecede42a7294f866a84bd3cabeb39fb0e/cpp/ql/lib/semmle/code/cpp/ir/implementation/aliased_ssa/IRBlock.qll#L254). Those predicates will all have the same contents and column order, so in practice any use of `hasIndexInBlock` will use the cached `getInstruction` directly.